Bindings/Java/OpenSimJNI/OpenSimContext.cpp

```diff
@@ -124,13 +125,21 @@ double OpenSimContext::getMuscleLength(Muscle& m) {
 }
 
 const Array<AbstractPathPoint*>& OpenSimContext::getCurrentPath(Muscle& m) {
-  return m.getGeometryPath().getCurrentPath(*_configState);
+  if (auto* p = dynamic_cast<const GeometryPath*>(&m.getPath())) {
```

Comment:
Agreed. Reverting these changes for now since I've added back `getGeometryPath()`. @aymanhab and I can discuss what makes sense for a refactor since (I believe) this file is GUI-specific.